pkg/provider/azure_local_services.go

```diff
@@ -218,3 +275,272 @@ func newBatchOperationResult(name string, success bool, err error) batchOperatio
 		err:     err,
 	}
 }
+
+func (az *Cloud) getLocalServiceInfo(serviceName string) (*serviceInfo, bool) {
+	data, ok := az.localServiceNameToServiceInfoMap.Load(serviceName)
+	if !ok {
+		return &serviceInfo{}, false
+	}
+	return data.(*serviceInfo), true
+}
+
+// setUpEndpointSlicesInformer creates an informer for EndpointSlices of local services.
+// It watches the update events and send backend pool update operations to the batch updater.
+// TODO (niqi): the update of endpointslice may be slower than tue update of endpoint pods. Need to fix this.
+func (az *Cloud) setUpEndpointSlicesInformer(informerFactory informers.SharedInformerFactory) {
+	endpointSlicesInformer := informerFactory.Discovery().V1().EndpointSlices().Informer()
+	_, _ = endpointSlicesInformer.AddEventHandler(
+		cache.ResourceEventHandlerFuncs{
+			AddFunc: func(obj interface{}) {
+				es := obj.(*discovery_v1.EndpointSlice)
+				az.endpointSlicesCache.Store(strings.ToLower(fmt.Sprintf("%s/%s", es.Namespace, es.Name)), es)
+			},
+			UpdateFunc: func(oldObj, newObj interface{}) {
+				previousES := oldObj.(*discovery_v1.EndpointSlice)
+				newES := newObj.(*discovery_v1.EndpointSlice)
+
+				svcName := getServiceNameOfEndpointSlice(newES)
+				if svcName == "" {
+					klog.V(4).Infof("EndpointSlice %s/%s does not have service name label, skip updating load balancer backend pool", newES.Namespace, newES.Name)
+					return
+				}
+
+				klog.V(4).Infof("Detecting EndpointSlice %s/%s update", newES.Namespace, newES.Name)
+				az.endpointSlicesCache.Store(strings.ToLower(fmt.Sprintf("%s/%s", newES.Namespace, newES.Name)), newES)
+
+				key := strings.ToLower(fmt.Sprintf("%s/%s", newES.Namespace, svcName))
+				si, found := az.getLocalServiceInfo(key)
```

Comment:
When user migrates from single LB to multiple LBs, a local svc may be not be found right? Will that cause any problem?

Reply:
No it won't, in this case the main loop will take over the process.